src/lib.rs

```diff
@@ -2439,6 +2456,38 @@ impl Build {
         Ok(())
     }
 
+    fn add_inherited_rustflags(&self, cmd: &mut Tool, target: &Target) -> Result<(), Error> {
+        let env_os = match self.getenv("CARGO_ENCODED_RUSTFLAGS") {
+            Some(env) => env,
+            // No encoded RUSTFLAGS -> nothing to do
+            None => return Ok(()),
+        };
+
+        let mut cc_flags: Vec<OsString> = env_os
+            .to_string_lossy()
+            .split("\u{1f}")
+            // Strip prefixes from rustc flags
+            .flat_map(|flag| {
+                if flag == "-Z" || flag == "-C" {
+                    None
+                } else if flag.starts_with("-Z") || flag.starts_with("-C") {
+                    Some(&flag[2..])
+                } else {
+                    Some(flag)
+                }
```

Comment:
I'm inclined to not support `-Z` flags? Or at least make it clear which ones are unstable

Reply:
The main motivation for this was aarch64 branch protection, because the way BTI for instance works is that if you link with any object that doesn't have BTI then it'll get disabled for the whole binary. The branch protection flags are still unstable in upstream Rust. We could just show a warning for those?

Reply:
The main motivation for this was aarch64 branch protection, because the way BTI for instance works is that if you link with any object that doesn't have BTI then it'll get disabled for the whole binary. The branch protection flags are still unstable in upstream Rust. We could just show a warning for those?

Reply:
Fair, maybe I'm just worried about these flags being treated the same by `cc` regardless of whether they're stable or not. Having different functions that parse them like `rustc_unstable_codegen_to_cflag` and `rustc_codegen_to_cflag` would appease a lot of my worries here.

Reply:
Right, so just do two passes where one matches `-C` flags and the other one matches `-Z` flags?

Reply:
I think I'd prefer that, yes